codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -147,4 +159,24 @@ private void constructHandler() {
                 "NewClientHandler", SmithyGoDependency.SMITHY_HTTP_TRANSPORT).build();
         writer.write("handler := $T($T(options.HTTPClient), stack)", decorateHandler, newClientHandler);
     }
+
+
+    /**
+     * Adds middleware to the operation middleware stack.
+     */
+    private void populateOperationMiddlewareStack() {
+        runtimeClientPlugins.forEach(runtimeClientPlugin -> {
+            if (!runtimeClientPlugin.matchesService(model, service)
+                    && !runtimeClientPlugin.matchesOperation(model, service, operation)) {
```

Comment:
This should be an or

Reply:
Here `and` is correct. By default when we do not define service/operation predicate on a runtime plugin, service predicate is set true whereas operation predicate is set to false. Having `or` would make generation return immaturely in case where service/operation predicate is not explicitly set.

when we set operation predicate for a runtime client plugin, service predicate is automatically turned to false. So I think `and` here makes sense. 

Alternatively will have to always define service/operation predicate explicitly at RuntimeClient plugin. I have confirmed the behavior when using `and` here and it works as expected.

Reply:
Ah okay I see the problem. Okay this is good for now. I think I'll need to fix up how those predicates work